cg/meta/workflow/balsamic.py

```diff
@@ -48,10 +64,267 @@ def link_sample(self, fastq_handler: BaseFastqHandler, sample: str, case: str):
+        """
+
+        if len(sample_data) == 0:
+            raise BalsamicStartError(f"{case_id} has no samples tagged for BALSAMIC analysis!")
+
+        arguments = {
+            "case_id": case_id,
+            "normal": self.get_verified_normal_path(sample_data=sample_data),
+            "tumor": self.get_verified_tumor_path(sample_data=sample_data),
+            "panel_bed": self.get_verified_bed(sample_data=sample_data, panel_bed=panel_bed),
+        }
+        return arguments
+
+    def get_sample_params(self, case_object) -> dict:
+        """Fetches attributes for each sample in given family, 
+        and returns them as a dictionary where SAMPLE ID is key"""
+        sample_data = {}
+        for link_object in self.get_balsamic_sample_objects(case_object):
+            sample_data[link_object.sample.internal_id] = {
+                "tissue_type": self.get_sample_type(link_object),
+                "concatenated_path": self.get_fastq_path(link_object),
+                "application_type": self.get_application_type(link_object),
+                "target_bed": self.get_target_bed_from_lims(link_object),
+            }
+        return sample_data
+
+    def report_sample_table(self, case_id: str, sample_data: dict):
+        """Outputs a table of samples to be displayed in log"""
+
+        LOG.info(f"Case {case_id} has following BALSAMIC samples:")
+        LOG.info(
+            "{:<20} {:<20} {:<20} {:<20}".format(
+                "SAMPLE ID", "TISSUE TYPE", "APPLICATION", "BED VERSION"
+            )
+        )
+        for key in sample_data:
+            LOG.info(
+                "{:<20} {:<20} {:<20} {:<20}".format(
+                    key,
+                    str(sample_data[key]["tissue_type"]),
+                    str(sample_data[key]["application_type"]),
+                    str(sample_data[key]["target_bed"]),
+                )
+            )
+        LOG.info("")
+
+    def parse_deliverables_report(self, deliverables_file_path, case_object) -> list:
+        """Parse BALSAMIC deliverables report, and return a list of files and their respective tags in bundle"""
+        sample_names = [x.sample.internal_id for x in self.get_balsamic_sample_objects(case_object)]
+        report_entries = dict(json.load(open(deliverables_file_path, "r")))["files"]
+        bundle_files = []
+        for entry in report_entries:
+            tags = entry["tags"].split(",")
+            for ind, tag in enumerate(tags):
+                for sample_name in sample_names:
+                    if sample_name in tag:
+                        tags[ind] = sample_name
+
+            bundle_file = {
+                "path": entry["path"],
+                "tags": tags,
+                "archive": False,
+            }
+            bundle_files.append(bundle_file)
+        return bundle_files
+
+    def update_housekeeper(self, case_object, sample_config, deliverables_file_path):
```

Comment:
I think this needs a more descriptive name, ditto for the update_statusdb

Reply:
Taken error message from log : number of tumor samples cannot be 0 to run balsamic
But I will add the whole string because thinking about it, there are other ways 0 can end up in log